osquery/events/eventfactory.h

```diff
@@ -0,0 +1,239 @@
+/**
+ * Copyright (c) 2014-present, The osquery authors
+ *
+ * This source code is licensed as defined by the LICENSE file found in the
+ * root directory of this source tree.
+ *
+ * SPDX-License-Identifier: (Apache-2.0 OR GPL-2.0-only)
+ */
+
+#pragma once
+
+#include <atomic>
+#include <functional>
```

Comment:
I wonder if we really need at these in the header?

Reply:
Probably not, I should have fixed the header inclusion earlier!